### PR TITLE
MEDME-7: Make changes github-webhook and pipeline to allow selecting the credentials to use for private repos

### PR DESCRIPTION
**Pull request by enyachoke**
https://mekomsolutions.atlassian.net/browse/MEDME-7

### CODE REVIEW

File: jenkins/jenkins_home/jobs/pipeline1/config.xml [hunk 1]
```diff
@@ -85,7 +89,8 @@ pipeline {
     }
     stage (&apos;Checkout the code and identify all Maven artifacts&apos;) {
       steps {
-        checkout scm: [$class: &apos;GitSCM&apos;, userRemoteConfigs: [[credentialsId: &apos;gitHubCredentials&apos;, url: repoUrl]], branches: [[name: branchName]]], poll: false
+        echo &quot;Creds=====&gt;${credentialsId} ${repoUrl} ${projectType}&quot;
```
Comment (rbuisson) [inline jenkins/jenkins_home/jobs/pipeline1/config.xml:92]: Let's keep the logging clean. It seems that by default, the `checkout` task will inform of the credential used:

```
...
The recommended git tool is: NONE
using credential simcgitlab
Cloning the remote Git repository
Cloning repository https://gitlab.tools.gbooking.ru/openmrs/ozone-simc.git
 > git init /var/jenkins_home/jobs/pipeline1/workspace # timeout=10
...
```

So we can probably just remove this logging line.

**Review by rbuisson — CHANGES_REQUESTED** [review]
Works good.
Just few remarks.
Plus, could you rename the job to "SCM Webhook"?
Or should we even say "Webhook"?

**Comment by rbuisson** [comment]
@enyachoke , could you add this to the README, after this line:
> The build is triggered by a Webhook that must be added to your hosting SCM service (GitHub, GitLab...).

The Webhook expects 2 query parameters:

- `[projectType=default|maven|distribution ...]`. Defaults to `default`.  See list of all supported project types [here](/Users/rom/repos/openmrs-cd/node-scripts/src/pipeline1/impl)

- `[credentialsId=<arbitrary_name_of_the_credentials_entry_in_Jenkins_to_clone_the_repo>]` . Defaults to `gitHubCredentials`.
Eg:
https://username:password@localhost:8080/generic-webhook-trigger/invoke?projectType=default&credentialsId=myGitLabRepoCredentials

File: jenkins/jenkins_home/jobs/github-webhook/config.xml [hunk 2]
```diff
@@ -86,15 +89,16 @@ projectType=$projectType
 repoUrl=$repoUrl
 branchName=$branchName
 commitId=$commitId
+credentialsId=${credentialsId-gitHubCredentials}
```
[the remark this answers is not in the document]
Comment (enyachoke) [reply jenkins/jenkins_home/jobs/github-webhook/config.xml:92]: The gitHubCredentials does not seem actually to work.

**Pull request merged**